libraries/botbuilder-dialogs-adaptive/tests/lgTelemetry.test.js

```diff
@@ -0,0 +1,51 @@
+
+
+const {Dialog } = require('botbuilder-dialogs');
+const assert = require('assert');
+
+const templateText = "test activity";
+const messageActivity = {
+    type: 'message',
+    text: 'test activity',
+    recipient: 'UK8CH2281:TKGSUQHQE',
+    from: 'BKGSYSTFG:TKGSUQHQE',
+    locale: 'en-us'
+};
+
+const lgTelemetryData = {
+    name : 'GeneratorResult',
+    properties: {
+        template : templateText,
+        result : messageActivity
+    }
+};
+
+class MockTelemetryClient {
+    constructor() {
+        this.telemetry = {};
+    }
+    
+    trackEvent(t) {
+        this.telemetry = t;
+    }
+}
+
+class MockDialog extends Dialog {
+    async beginDialog(dc, options) {
+        this.telemetryClient.trackEvent(lgTelemetryData);
+    }
+}
+
+describe('ActivityFactoryTest', function() {
+
+    it('basic telemetry test', () => {
+        const mockDialog = new MockDialog();
+        mockDialog.telemetryClient = new MockTelemetryClient();

```

Comment:
Since a Dialog will automatically have at least a NullTelemetryClient, you can use `sinon` to stub out the `trackEvent` method on `dialogInstance.telemetryClient`.

```js
const { stub } = require('sinon');
const trackEventStub = stub(mockDialog.telemetryClient, 'trackEvent');

// Later after the method that calls trackEvent finishes
assert(trackEventStub.called);
```

And use a captureAction to verify the stub's input.

There are also examples in [botframeworkAdapter.test.js](https://github.com/microsoft/botbuilder-js/blob/68a52a2e4069ece47fa8a8760caf2c2968c9765b/libraries/botbuilder/tests/botFrameworkAdapter.test.js#L297-L302) and the [skillDialog.test.js](https://github.com/microsoft/botbuilder-js/blob/68a52a2e4069ece47fa8a8760caf2c2968c9765b/libraries/botbuilder-dialogs/tests/skillDialog.test.js#L140-L156) files. The SkillDialog tests may be more helpful and will show an example closer to what @boydc2014 suggested. Additionally, the tests for SkillDialog in JS are _extremely_ close to the tests in .NET (see [SkillDialogTests.cs](https://github.com/microsoft/botbuilder-dotnet/blob/4915c3df6f9c3ceb7032f17eb10b5242989fa487/tests/Microsoft.Bot.Builder.Dialogs.Tests/SkillDialogTests.cs#L105)), so that may help in authoring tests for the LG Telemetry changes.

Reply:
I have added some telemetry tests in skillDialog.test.js. Please take a look. Thanks.

Reply:
@Danieladu the updates to the `BeginSkill` tests looks great, can you move `createTelemetryClientAndStub()` to a utilities file since that capture action is self contained and then use it for the other tests as well?